### PR TITLE
Ability to directly move an item to another directory

### LINKED ISSUE #946: Ability to directly move an item to another directory
Please add an option to directly move an item to another directory similar to the "Move to" option in File Explorer's ribbon.

### PR DESCRIPTION
**Pull request by oleitao**
**Details of Changes**
Add details of changes here.
- Added the ability to directly move an item to another directory

Closes #946

**Validation**
How did you test these changes?
- [x] Select an folder and move into another folder

test1

### CODE REVIEW

**Comment by yaichenbaum** [comment]
@oleitao It looks like the screen recording got messed up, can you try recording it again?

**Comment by oleitao** [comment]
> @oleitao It looks like the screen recording got messed up, can you try recording it again?

@yair100 ofcourse i can

test1

**Comment by yaichenbaum** [comment]
@oleitao Can you link to an issue?

**Comment by oleitao** [comment]
> @oleitao Can you link to an issue?

@yair100 i don't have permission to link my PR

**Comment by yaichenbaum** [comment]
@oleitao You can add it to the description https://github.com/blog/957-introducing-issue-mentions.

**Comment by oleitao** [comment]
> @oleitao You can add it to the description https://github.com/blog/957-introducing-issue-mentions.

I fixed issue in [#946](https://github.com/files-community/Files/issues/946)

**Comment by yaichenbaum** [comment]
@oleitao You can add `Closes #946` to the PR description to link to the issue.

**Comment by oleitao** [comment]
> @oleitao You can add `Closes #946` to the PR description to link to the issue.

How do i do that?

**Comment by yaichenbaum** [comment]
> > @oleitao You can add `Closes #946` to the PR description to link to the issue.
> 
> How do i do that?

I just added it for you.
image

**Comment by oleitao** [comment]
> > > @oleitao You can add `Closes #946` to the PR description to link to the issue.
> > 
> > 
> > How do i do that?
> 
> I just added it for you.
> image

**Details of Changes**

- Added on right-click ContextMenu an option command 'Move...' to move from an folder to another selected by user.

Closes [#946](https://github.com/files-community/Files/issues/946)

**Validataion**

- Select an folder to move into another destination
- Select 'Move...' option
- Select on DialogBox destination folder to move
- Navigate into destination folder and check if selected folder was correctly moved (and it's content)
- Navigate into origin folder and check if selected folder doesn't exist.

Capture1

**Comment by oleitao** [comment]
@yair100 i would like to resolve conflicts in order to finish this PR. However, i'm not allowed to do that
Capture3

**Comment by yaichenbaum** [comment]
@oleitao That's because there are too many conflicts to resolve through GitHub, you'll need to resolve them locally.

File: Files/Interacts/BaseLayoutCommandImplementationModel.cs [hunk 1]
```diff
@@ -564,9 +564,14 @@ public void SearchUnindexedItems(RoutedEventArgs e)
             associatedInstance.SubmitSearch(associatedInstance.InstanceViewModel.CurrentSearchQuery, true);
         }
 
-        public async void CreateFolderWithSelection(RoutedEventArgs e)
+        public virtual async void MoveItem(RoutedEventArgs e)
         {
-            UIFilesystemHelpers.CreateFolderWithSelectionAsync(associatedInstance);
```
Comment (tsvietOK) [inline Files/Interacts/BaseLayoutCommandImplementationModel.cs:569]: Why did you remove `CreateFolderWithSelection` method?

**Comment by tsvietOK** [comment]
Looks like this function is not working with shortcut files, so i guess, `Move` menu item shouldn't be visible for shortcuts.

**Comment by tsvietOK** [comment]
When pressing Move button app crashes.
By the way, to hide Move menu item for shortcuts, just use the next:
```cs
{
    Text = "BaseLayoutItemContextFlyoutMove/Text".GetLocalized(),
    Glyph = "\uE8B4",
    Command = commandsViewModel.MoveItemCommand,
    ShowItem = selectedItems.All(x => !x.IsShortcutItem)
},

```
Please, test app properly before requesting review.

**Comment by yaichenbaum** [comment]
Closing this PR due to inactivity.
